### PR TITLE
feat(useTimeout): add pause and resume methods 

### PR DESCRIPTION
**Pull request by marcoarruda**
### Before submitting the PR, please make sure you do the following

- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [X] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [X] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

<details>
<summary><strong>⚠️ Slowing down new functions</strong></summary>
<br>

> **Warning**: **Slowing down new functions**
>
> As the VueUse audience continues to grow, we have been inundated with an overwhelming number of feature requests and pull requests. As a result, maintaining the project has become increasingly challenging and has stretched our capacity to its limits. As such, in the near future, we may need to slow down our acceptance of new features and prioritize the stability and quality of existing functions. **Please note that new features for VueUse may not be accepted at this time.** If you have any new ideas, we suggest that you first incorporate them into your own codebase, iterate on them to suit your needs, and assess their generalizability. If you strongly believe that your ideas are beneficial to the community, you may submit a pull request along with your use cases, and we would be happy to review and discuss them. Thank you for your understanding.

</details>

---

### Description

Implements Pausable controls on `useTimeout` and `useTimeoutFn` as described here: #3807

demo-timeout-pause

### Additional context

Any optimization on internal variables used to control Pausable feature

### CODE REVIEW

**Comment by aparajita** [comment]
@marcoarruda Will the change in return type break any existing code?

**Comment by marcoarruda** [comment]
> @marcoarruda Will the change in return type break any existing code?

Hi @aparajita , I just pushed an adjustment to support the MaybeRef interval.
Some axios tests are failing locally, I don't think it's related though
Screenshot 2024-03-26 at 11 20 09

File: packages/shared/useTimeoutFn/index.ts [hunk 1]
```diff
@@ -24,13 +25,17 @@ export function useTimeoutFn<CallbackFn extends AnyFn>(
...
+): Stoppable<Parameters<CallbackFn> | []> & Pausable & { timeLeft: ShallowRef<number> } {
   const {
     immediate = true,
   } = options
 
   const isPending = ref(false)
 
+  const isActive = ref(false)
+  const startedAt = ref(-1)
+  const timeLeft = ref(-1)
```
Comment (OrbisK) [inline packages/shared/useTimeoutFn/index.ts:37]: I think this should not default to `-1` `new Date(-1)` is valid and `Thu Jan 01 1970 00:59:59`

**Comment by antfu** [comment]
Thank you for working on this, but I am afraid this might go beyond the scope of this composable. `useTimeout` is designed to be a simple wrapper around `setTimeout` - where it's not an accurate timer. What you are looking for is probably better to be handled by a separate package with proper timer implementation.

**Pull request closed without merging**